libtsuba/src/RDGManifest.h

```diff
@@ -0,0 +1,151 @@
+
+  //
+  // Persisted
+  //
+  uint64_t version_{0};
+  uint64_t previous_version_{0};
+  uint32_t num_hosts_{0};  // 0 is a reserved value for the empty RDG when
+  // tsuba views policy_id as zero (not partitioned) or not zero (partitioned
+  // according to a CuSP-specific policy)
+  uint32_t policy_id_{0};
+  bool transpose_{false};
+  RDGLineage lineage_;
+  std::string view_type_;
+  std::vector<std::string> view_args_;
+
+  RDGManifest(katana::Uri dir)
+      : dir_(std::move(dir)), view_type_(gDefaultRDGViewType) {}
+  RDGManifest(katana::Uri dir, const std::string& view_type)
+      : dir_(std::move(dir)) {
+    view_type_ = gDefaultRDGViewType;
+    if (view_type != "") {
+      view_type_ = view_type;
+    }
+  }
+
+  RDGManifest(
+      uint64_t version, uint64_t previous_version, uint32_t num_hosts,
+      uint32_t policy_id, bool transpose, katana::Uri dir, RDGLineage lineage)
+      : dir_(std::move(dir)),
+        version_(version),
+        previous_version_(previous_version),
+        num_hosts_(num_hosts),
+        policy_id_(policy_id),
+        transpose_(transpose),
+        lineage_(std::move(lineage)) {}
+
+  static katana::Result<RDGManifest> MakeFromStorage(const katana::Uri& uri);
+
+  static std::string PartitionFileName(
+      const std::string& view_type, uint32_t node_id, uint64_t version);
+
+  std::string view_specifier() const {
+    if (view_args_.size())
+      return fmt::format("{}-{}", view_type_, fmt::join(view_args_, "-"));
+    return view_type_;
+  }
+
+public:
+  RDGManifest() = default;
+
+  RDGManifest NextVersion(
+      uint32_t num_hosts, uint32_t policy_id, bool transpose,
+      const RDGLineage& lineage) const {
+    return RDGManifest(
+        version_ + 1, version_, num_hosts, policy_id, transpose, dir_, lineage);
+  }
+
+  bool IsEmptyRDG() const { return num_hosts() == 0; }
+
+  static katana::Result<RDGManifest> Make(RDGHandle handle);
+
+  /// Create an RDGManifest
+  /// \param uri a uri that either names a registered RDG or an explicit
+  ///    rdg file
+  /// \returns the constructed RDGManifest and the directory of its contents
+  static katana::Result<RDGManifest> Make(const katana::Uri& uri);
+
+  /// Create an RDGManifest
+  /// \param uri is a storage prefix where the RDG is stored
+  /// \param view_type is the view_type of the desired RDG
+  /// \param version is the version of the meta_dir to load
+  /// \returns the constructed RDGManifest and the directory of its contents
+  static katana::Result<RDGManifest> Make(
+      const katana::Uri& uri, const std::string& view_type, uint64_t version);
+
+  const katana::Uri& dir() const { return dir_; }
+  uint64_t version() const { return version_; }
+  uint32_t num_hosts() const { return num_hosts_; }
+  uint32_t policy_id() const { return policy_id_; }
+  uint64_t previous_version() const { return previous_version_; }
+  const std::string& get_viewtype() const { return view_type_; }
```

Comment:
Small style nit. Per our guidelines this would be either `viewtype()` or `GetViewType()`. I prefer the first (in general, but here especially) because it matches the others in this class.

https://google.github.io/styleguide/cppguide.html#Function_Names

Reply:
Yeah, this function returns all views and makes no assumptions regarding the users intentions for requesting the available view list.